### PR TITLE
Remove the dependency on github.com/prometheus/prometheus

### PR DESCRIPTION
**Pull request by skhalash**
## Description

Changes proposed in this pull request:

- Replace the config  github.com/prometheus/prometheus with a minimalistic self-written Prometheus config
- Adapt the unit tests, we don't need to check dummy empty values anymore

## Related Issues and Documents

Related issues:
https://github.com/kyma-project/telemetry-manager/issues/318

## Traceability

- [x] The PR is linked to a GitHub Issue.
- [ ] New features have a milestone label set.
- [x] New features have defined acceptance criteria in a corresponding GitHub Issue, and all criteria are satisfied with this PR.
- [x] The corresponding GitHub Issue has a respective `area` label.
- [ ] The follow-up issues (if any) are linked in the `Related Issues` section.

## Testability

The feature is unit-tested:

- [x] Yes.
- [ ] No, because unit tests are not needed.
- [ ] No, because of ...

The feature is e2e-tested:

- [x] Yes.
- [ ] No, because e2e-tests are not needed.
- [ ] No, because of ...

Tests conducted for the PR:

## Codebase

- [x] My code follows the [Effective Go](https://go.dev/doc/effective_go) style guidelines.
- [x] The code was planned and designed following the defined architecture and the separation of concerns.
- [ ] The code has sufficient comments, particularly for all hard-to-understand areas.
- [x] This PR adds value and shows no feature creep.
- [x] I have augmented the test suite that proves my fix is effective or that my feature works.
- [ ] Adjusted the documentation if the change is user-facing.

### CODE REVIEW

File: internal/otelcollector/config/metric/agent/receivers.go [hunk 1]
```diff
@@ -47,25 +39,16 @@ func makeKubeletStatsConfig() *KubeletStatsReceiver {
...
-
 	return &PrometheusReceiver{
-		Config: promconfig.Config{
-			ScrapeConfigs: []*promconfig.ScrapeConfig{
+		Config: PrometheusConfig{
+			ScrapeConfigs: []ScrapeConfig{
 				{
-					JobName:          "opentelemetry-collector",
-					ScrapeInterval:   prommodel.Duration(10 * time.Second),
-					HTTPClientConfig: promcommonconfig.DefaultHTTPClientConfig,
```
Comment (rakesh-garimella) [inline internal/otelcollector/config/metric/agent/receivers.go:66]: is this enabled by defualt so this is removed ?

Comment (skhalash) [reply internal/otelcollector/config/metric/agent/receivers.go:66]: Yes, those are default values

**Review by rakesh-garimella — APPROVED** [review]

**Review by chrkl — APPROVED** [review]

**Review by NHingerl — APPROVED** [review]

**Pull request merged**